painttiming.bs

```diff
@@ -62,9 +83,25 @@ Formally, we consider the user agent to have "rendered" a document when it has p
 
     NOTE: The rendering pipeline is very complex, and the timestamp should be the latest timestamp the user agent is able to note in this pipeline (best effort). Typically the time at which the frame is submitted to the OS for display is recommended for this API.
 
+<dfn>Contentful</dfn>: An element counts as contentful when one or more of the following apply:
+* The element has a [=text node=] descendant, with at least one character, excluding [=document white space characters=].
+* The element is a [=replaced image=] representing a [=loaded=] image.
+* The element's current style contains a [=loaded=] [=background-image=] or [=mask-image=].
+* The element is a [=canvas=] with its [=context mode=] set to any value other than <code>none</code>.
+* The element is an [=svg element with rendered descendants=].
+
+<dfn>Paintable</dfn>: An element counts as paintable when all of the following apply:
+* The element intersects with the document, as defined by [=Intersection rect algorithm=]. 
+
+    NOTE: This covers the cases where the element is transformed to zero size, or is positioned in negative coordinates,
+
+* The element's [=computed=] [=visibility=] is <code>visible</code>.
+* The element is not fully transparent, as defined  [=computed=] [=opacity=] is greater than 0.
```

Comment:
Does this account for opacity on ancestors?

Reply:
This doesn't, the next line should (all the ancestors should have a computed opacity greather than 0).